vent/logger/logger.py

```diff
@@ -0,0 +1,197 @@
+        else:
+            self.control_table = self.h5file.root.controls.readout
+            
+    def close_logfile(self):
+        """
+        Flushes & closes the open hdf file.
+        """
+        self.h5file.close() # Also flushes the remaining buffers
+
+    def __rescale_save(self, raw_value, value, target, scalingconstant):
+        """
+        Rescales values to fit into the ranges specified in DataSample
+        """
+        if DataSample.columns[value].dtype == target.dtype:
+            return np.int8( raw_value * scalingconstant )    # type cast to unsigned int 8, flow between 0 and 1.28 l/sec
+        else:
+            raise ValueError('Dynamic range unclear.')
+
+    def store_waveform_data(self, sensor_values: SensorValues, control_values: ControlValues):
+        """
+        Appends a datapoint to the file.
+        NOTE: Not flushed yet.
+        """
+        self._open_logfile()
+        datapoint                 = self.data_table.row
+        datapoint['timestamp']    = sensor_values.timestamp
+        datapoint['pressure']     = self.__rescale_save(raw_value = sensor_values.PRESSURE,  value = 'pressure', target=pytb.UInt8Col(), scalingconstant=5)
+        datapoint['cycle_number'] = sensor_values.breath_count
+        datapoint['flow_in']      = self.__rescale_save(raw_value = control_values.flow_in,  value = 'flow_in',  target=pytb.UInt8Col(), scalingconstant=100)
+        datapoint['flow_out']     = self.__rescale_save(raw_value = control_values.flow_out, value = 'flow_out', target=pytb.UInt8Col(), scalingconstant=100)
+        datapoint.append()
+    
+    def store_control_command(self, control_setting: ControlSetting):
+        """
+        Appends a control signal to the hdf5 file.
+        NOTE: Also not flushed yet.
+        """
+        self._open_logfile()
+        datapoint               = self.control_table.row
+        datapoint['name']       = control_setting.name
+        datapoint['value']      = control_setting.value
+        datapoint['min_value']  = control_setting.min_value
+        datapoint['max_value']  = control_setting.max_value
+        datapoint['timestamp']  = control_setting.timestamp
+        datapoint.append()
+
+    def flush_logfile(self):
+        """
+        This flushes the datapoints to the file.
+        To be executed every other second, e.g. at the end of breath cycle.
+        """
+        self.data_table.flush()
+                    
+    def check_files(self):
+        """
+        make sure that the file's are not getting too large.
+        """
+        total_size = 0
+        logpath = 'vent/logfiles'
+        for filenames in os.listdir(logpath):
+            fp = os.path.join(logpath, filenames)
+            # skip if it is symbolic link
+            if not os.path.islink(fp):
+                total_size += os.path.getsize(fp)
+
+        if len(os.listdir(logpath)) > 1000:
```

Comment:
Can raise the exception together with the exact space number used. E.g. "warning: used 11.5GB"

Reply:
good suggestion, done!